Cabal/src/Distribution/Simple/Configure.hs

```diff
@@ -734,7 +735,7 @@ computeLocalBuildConfig cfg comp programDb = do
             CompilerId GHC _ ->
               -- if ghc is dynamic, then ghci needs a shared
               -- library, so we build one by default.
-              GHC.isDynamic comp
+              GHC.compilerBuildWay comp == DynWay
```

Comment:
Is the logic consistent here? If we have `--enable-executable-dynamic --enable-profiling`, do we want `sharedLibsByDefault` to be `True`?
Conversely, why don't we have `GHC.compilerBuildWay comp `elem` [DynWay, ProfDynWay]`? What will happen with `ProfDyn` and ghci? Is that supported?

Reply:
`sharedLibsByDefault` is supposed to turn on building dynamic libraries when you have a dynamically linked ghc so template haskell works.